cmd/collectors/zapi/collector/zapi.go

```diff
@@ -251,15 +252,23 @@ func (z *Zapi) PollData() (map[string]*matrix.Matrix, error) {
 		oldInstances.Add(key)
 	}
 
-	fetch = func(instance *matrix.Instance, node *node.Node, path []string) {
+	fetch = func(instance *matrix.Instance, node *node.Node, path []string, isAppend bool) {
 
 		newpath := append(path, node.GetNameS())
 		key := strings.Join(newpath, ".")
 		z.Logger.Trace().Msgf(" > %s(%s)%s <%s%d%s> name=[%s%s%s%s] value=[%s%s%s]", color.Grey, newpath, color.End, color.Red, len(node.GetChildren()), color.End, color.Bold, color.Cyan, node.GetNameS(), color.End, color.Yellow, node.GetContentS(), color.End)
+
 		if value := node.GetContentS(); value != "" {
 			if label, has := z.instanceLabelPaths[key]; has {
-				instance.SetLabel(label, value)
-				z.Logger.Trace().Msgf(" > %slabel (%s) [%s] set value (%s)%s", color.Yellow, key, label, value, color.End)
+				// Handling array with comma separated values
+				previousValue := instance.GetLabel(label)
+				if isAppend && previousValue != "" {
+					instance.SetLabel(label, previousValue+","+value)
```

Comment:
Out of 4 volume plugin(zapi, zapiperf, rest, restperf), 
- zapiperf and restperf are exact same and handling `aggr`, `node` as perf call don't have this array in response. --> no changes required.
- zapi don't have any handling for flexgroup.
- rest has this below handling for `aggruuid` and it still holds true as zapi don't handle aggruuid for flexgroup.
<img width="712" alt="image" src="https://user-images.githubusercontent.com/83282894/225631329-529b00aa-8d51-47c5-8b0a-ed2b4ba030c8.png">

Reply:
So, Zapi would handle this way, 
First populate `aggrsDisksMap` and then populate `aggrsMap`.
For flexgroup, which span across multiple aggregates, it can't be part of any map. 
So for them label `isHardwareEncrypted` alway be false.
```
for _, aggrDiskData := range result {
		aggrUUID := aggrDiskData.GetChildContentS("aggregate-uuid")
		aggrName := aggrDiskData.GetChildContentS("aggregate")
		aggrDiskList := aggrDiskData.GetChildS("aggr-plex-list").GetChildS("aggr-plex-info").GetChildS("aggr-raidgroup-list").GetChildS("aggr-raidgroup-info").GetChildS("aggr-disk-list").GetChildren()
		for _, aggrDisk := range aggrDiskList {
			diskName = aggrDisk.GetChildContentS("disk")
			aggrsDisksMap[diskName] = aggrData{aggrUUID: aggrUUID, aggrName: aggrName}
		}
	}
	return aggrsDisksMap, nil
```
```
func (my *Volume) updateAggrMap(disks []string, aggrDiskMap map[string]aggrData) {
	if disks != nil && aggrDiskMap != nil {
		// Clean aggrsMap map
		my.aggrsMap = make(map[string]string)

		for _, disk := range disks {
			aggr := aggrDiskMap[disk]
			my.aggrsMap[aggr.aggrUUID] = aggr.aggrName
		}
	}
}
```

<img width="785" alt="image" src="https://user-images.githubusercontent.com/83282894/226327378-6e890f9b-fd1a-4572-a35a-3135965f372f.png">

<img width="1774" alt="image" src="https://user-images.githubusercontent.com/83282894/226328105-4973f3bc-d6f1-4dbc-8da6-4d9e6110b141.png">